spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/BaseDeleteOrphanFilesSparkAction.java

```diff
@@ -220,6 +222,7 @@ private DeleteOrphanFiles.Result doExecute() {
     Column joinCond = nameEqual.and(actualContains);
     List<String> orphanFiles = actualFileDF.join(validFileDF, joinCond, "leftanti")
         .as(Encoders.STRING())
+        .unpersist()
```

Comment:
I do not think this will trigger the unpersist on "allManifests"

Reply:
yeah, I figured that out, but didn't figured out the final solution. So, changed to draft :) 

But as you said, I need to call allManifests table again I guess.

Reply:
@RussellSpitzer , which is better? a cache or manually calling dataset.collect() on that allManifest df and building new dataset on top of those collected rows and reusing it on both the location?